docs/source/developers/reviewing.rst

```diff
@@ -255,3 +255,43 @@ Social aspects
 * Like any communication, code reviews are governed by the Apache
   `Code of Conduct <https://www.apache.org/foundation/policies/conduct.html>`_.
   This applies to both reviewers and contributors.
+
+
+Labelling
+=========
+
+While reviewing PRs, we should try to identify whether the corresponding issue 
+needs to be marked with one or both of the following issue labels:
+
+* **Critical Fix**: The change fixes either: (a) a security vulnerability;
+  (b) a bug that causes incorrect or invalid data to be produced;
+  or (c) a bug that causes a crash (while the API contract is upheld).
+  This is intended to mark fixes to issues that may affect users without their
+  knowledge. For this reason, fixing bugs that cause errors don't count, since 
+  those bugs are usually obvious. Bugs that cause crashes are considered critical
+  because they are a possible vector of Denial-of-Service attacks.
+* **Breaking Change**: The change breaks backwards compatibility in a public API.
+  For changes in C++, this does not include changes that simply break ABI
+  compatibility, except for the few places where we do guarantee ABI
```

Comment:
Why do we not include ABI compatibility?
I think that we need to bump major version when we break API or ABI compatibility.
I thought that we use "Breaking Change" to determine whether we should bump major/minor version.

BTW, we will be able to check ABI compatibility automatically by CI. (Some projects do it.) So we may not check ABI compatibility manually.

Reply:
What I meant is that we don't guarantee that your code compiled against the headers of Arrow 10.0.0 can be linked against the Arrow 11.0.0 shared libraries. But we do try to keep it stable such that you don't need to change your code when you upgrade the dependency.

For example, IIUC changing a function signature from

```cpp
void MyFunction(int arg1);
```

To:

```cpp
void MyFunction(int arg1, int arg2 = 0);
```

Breaks ABI compatibility, since it alters the symbol for `MyFunction` in the shared library. But downstream projects won't have to change their code when they upgrade. So this paragraph means we wouldn't label this change as a "Breaking Change".

Reply:
Thanks. I see.

My understanding: We use "Breaking Change" to provide upgrade difficulty for users. We don't use "Breaking Change" to determine which version component (major/minor/patch) should be bumped.

Reply:
For now, yes.

Reply:
> We don't use "Breaking Change" to determine which version component (major/minor/patch) should be bumped.

I suspect this won't be viable until we do https://github.com/apache/arrow/issues/15280, but worth discussing after.